### PR TITLE
openlineage wrapper for spark

### PR DESCRIPTION
**Pull request by OleksandrDvornik**
Add Spark subclasses to extend OpenLineageSparkListener and javaagent from OpenLineage #1536

### CODE REVIEW

**Comment by OleksandrDvornik** [comment]
@collado-mike  I've tested your suggestion for using SparkAgent directly - it works! So, I've removed MarquezAgent class also

**Review by wslulciuc — CHANGES_REQUESTED** [review]
@OleksandrDvornik ~We don't want to remove the code for the Spark agent, but rather just deprecate (see https://github.com/MarquezProject/marquez/pull/1545). Meaning, we'll want to remove the spark integration from the root [build.gradle](https://github.com/MarquezProject/marquez/blob/main/build.gradle) file to no longer hook into the project build. I'd prefer we write a simple Spark app to test the integration with OpenLineage in CI with a running instance of Marquez to assert the metadata collected. We'll still want to fetch the snapshot from jfrog, but the Spark app would be its own separate project for testing.~

**Review by collado-mike — APPROVED** [review]
Per offline discussion, we'll wait for the OpenLineage release before merging this

**Review by wslulciuc — APPROVED** [review]
NIICE! ❤️

File: integrations/spark/build.gradle
Comment (wslulciuc) [inline integrations/spark/build.gradle:36]: Mind updating the `openlineage-spark` dep to `0.1.0`?

File: integrations/spark/build.gradle
Comment (wslulciuc) [inline integrations/spark/build.gradle:155]: Since this jar will still be published by Marquez, we'll want the keep `marquez.*`

File: integrations/spark/src/main/resources/META-INF/MANIFEST.MF
Comment (wslulciuc) [inline integrations/spark/src/main/resources/META-INF/MANIFEST.MF:3]: Do we need this file? We already define the manifest in [`spark/build.gradle`](https://github.com/MarquezProject/marquez/blob/main/integrations/spark/build.gradle#L165)

File: integrations/spark/build.gradle
Comment (OleksandrDvornik) [reply integrations/spark/build.gradle:155]: we can't change that, cause we don't have agent in marquez, we have only listener. This is an entry for javaagent, so it will try to find such class. By Mike comment, I've removed that SparkAgent from marquez code.

File: integrations/spark/src/main/resources/META-INF/MANIFEST.MF
Comment (OleksandrDvornik) [reply integrations/spark/src/main/resources/META-INF/MANIFEST.MF:3]: I guess not, testing that

Comment (OleksandrDvornik) [reply integrations/spark/src/main/resources/META-INF/MANIFEST.MF:3]: we don't need it, gradle will create it for us.

File: integrations/spark/build.gradle
Comment (wslulciuc) [reply integrations/spark/build.gradle:155]: Ahh, makes sense 👍

File: integrations/spark/src/main/resources/META-INF/MANIFEST.MF
Comment (wslulciuc) [reply integrations/spark/src/main/resources/META-INF/MANIFEST.MF]: Nice 👍

**Pull request merged**
